pkg/plugins/backendplugin/grpcplugin/client_proto.go

```diff
@@ -66,57 +80,80 @@ func (r *protoClient) Logger() log.Logger {
 }
 
 func (r *protoClient) Start(ctx context.Context) error {
-	return r.plugin.Start(ctx)
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	err := r.plugin.Start(ctx)
+	if err != nil {
+		return err
+	}
+
+	r.running = true
+	return nil
 }
 
 func (r *protoClient) Stop(ctx context.Context) error {
-	return r.plugin.Stop(ctx)
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	err := r.plugin.Stop(ctx)
```

Comment:
Nit. Are this setting `r.plugin.pluginClient` to nil? If so, maybe the running flag not needed?